lib/files.ex

```diff
@@ -33,12 +33,10 @@ defmodule Uploadex.Files do
     files = wrap_files(record, uploader)
     extensions = get_accepted_extensions(record, uploader)
 
-    case Validation.validate_extensions(files, extensions) do
-      :ok ->
-        files
-        |> Enum.filter(fn {file, _, _} -> is_map(file) end)
-        |> do_store_files(record)
+    changed_files = Enum.filter(files, fn {file, _, _} -> is_map(file) end)
```

Comment:
I'm not sure this would be the expected behaviour, because of what the doc says:
> Files that are not maps are ignored, which allows for assigning an existing file to a record without recreating it, by simply passing it's filename.

So not being a map doesn't mean it's an old file, you can also create/update a record providing just a filename, which could be an URL, for example. In this case, I would still expect the extension validation to be executed. What do you think?

Reply:
Your'e right! So for this to work we will need to get the changed files from the `Changeset`, right? Looking at the code seems that it will need a lot of changes, and possibly breaking changes for the `store_files/2` function, so better not alter this now, right?

Or there's a simpler way of doing this?

Reply:
There is a simpler way:

In `Context.update_with_file`, we could provide the `previous_record` to `store_files`, which would accept a `previous_record` as an optional argument. If provided, `store_files` would check which files in the `new_files` are not in `old_files` and only validate those.

Important to have `store_files` accept this new argument as optional and not change the order of arguments to not do a breaking change.

What do you think?

Reply:
I think this will work nicely! :)